src/core/core.scale.js

```diff
@@ -287,8 +288,15 @@ module.exports = Element.extend({
 		helpers.callback(this.options.beforeBuildTicks, [this]);
 	},
 	buildTicks: helpers.noop,
-	afterBuildTicks: function() {
-		helpers.callback(this.options.afterBuildTicks, [this]);
+	afterBuildTicks: function(ticks) {
+		var me = this;
+		// ticks is empty for old axis implementations here
+		if (ticks.length) {
```

Comment:
Could `ticks` be undefined in some cases? Maybe better to check `ticks && ticks.length` or `helpers.isArray(ticks)`?

Reply:
`ticks = me.buildTicks() || [];`
But I guess it could be something other than array still.

Reply:
I was thinking about external scales calling `buildTicks()`